### ISSUE TITLE
ERROR 500 WHEN TRYING TO EXTRACT DATA

### ISSUE THREAD

**Issue by samapera3478** [tabulapdf/tabula#908]
![tabula error](https://user-images.githubusercontent.com/30800171/44206684-367d3200-a120-11e8-8922-475f3edb6e49.PNG)

[063-01X-8096-01_R1_WM0.pdf](https://github.com/tabulapdf/tabula/files/2293564/063-01X-8096-01_R1_WM0.pdf)

[063-01X-0106-01_R1_WM0.pdf](https://github.com/tabulapdf/tabula/files/2293573/063-01X-0106-01_R1_WM0.pdf)

I have been able to extract the data from the table on the right of these type of drawings before but I was trying to do it on a new batch and it kept giving me this error. I was able to find the first file that is cause the issue (attached - 063-01X-8096-01) but I can't see the difference between that corrupt file and the others, thus I can't extract them and there are too many to try one by one. I also attached one drawing that works (063-01X-0106-01)

**Comment by rosenjcb** [comment]
I've had this happen to me as well. I've currently spent the whole day trying to debug this error:
`(NoMethodError) undefined method `[]' for nil:NilClass`

Is there a way to get a more detailed stacktrace?

**Comment by jeremybmerrill** [comment]
@rosenjcb You'll want to [run Tabula from source](https://github.com/tabulapdf/tabula#running-tabula-from-source-for-developers) to get the full stacktrace. Once you get it, please share it here; an error like you include above shouldn't be happening at all.

**Comment by rosenjcb** [comment]
@jeremybmerrill I actually build both tabula and tabula-java from source (I've forked each repo). I get the attached image as my error and nothing in the jetty console besides `***[{"page":1,"extraction_method":"guess","selection_id":"F1536100935342","x1":44.94599395751953,"x2":772.5959939575196,"y1":140.085,"y2":262.84499999999997,"width":727.65,"height":122.76}]!!!!!DEBUG: resetting rack response due exception`

![errorundefined](https://user-images.githubusercontent.com/8102129/45104376-d528fd00-b0e6-11e8-8050-cce6b3016962.png)

**Comment by jeremybmerrill** [comment]
Right, if you run it from source with `jruby -G -r jbundler -S rackup` rather than the Jetty / warbler stuff, you'll get a real stacktrace in the console. (That `jruby` command is what I use whenever I'm developing on Tabula; the warbler stuff is a far more complicated way of packing the app up for distribution which involves compromises -- like less visibility of error messages.)

**Comment by rosenjcb** [comment]
@jeremybmerrill It worked! That is a much better stacktrace than what I was getting before.  I was running the server from jar. After reading the instructions again, it seems that running `
java -jar build\tabula.jar` is for testing the package when production ready. From now on I'll use the rackup command when developing and debugging. Thank you.

**Comment by jeremybmerrill** [comment]
Super. If you have recommendations on how to make the README clearer, I'm all ears. LIkewise if the bug you find is rooted in mainline Tabula code (as opposed to y'all's modifications).

**Comment by samapera3478** [comment]
Does this mean the error is fixed now?

**Comment by rosenjcb** [comment]
@samapera3478 No, it just means I co-opted your thread. You should try compiling the project from source and running the `jruby -G -r jbundler -S rackup` command for a better stacktrace. Then comeback and let us know what error you see.

**Comment by samapera3478** [comment]
I'm sorry I'm not a developer, I have no clue what that means. I just use
it to extract some info from drawings to make our lives a little easier
